library/src/main/java/com/datatorrent/lib/db/jdbc/JdbcPOJOInsertOutputOperator.java

```diff
@@ -146,28 +152,36 @@ private String getMatchingField(Field[] fields, String columnName)
     return null;
   }
 
-  protected void populateColumnDataTypes(String columns) throws SQLException
+  /**
+   * Function to populate Meta Data.
+   * @param columnNamesSet
+   * @throws SQLException
+   */
+  protected void populateColumnDataTypes(HashSet<String> columnNamesSet) throws SQLException
   {
-    columnNames = Lists.newArrayList();
-    columnDataTypes = Lists.newArrayList();
-    columnNullabilities = Lists.newArrayList();
-
-    try (Statement st = store.getConnection().createStatement()) {
-      if (columns == null || columns.length() == 0) {
-        columns = "*";
+    ResultSet rsColumns;
+    DatabaseMetaData meta = store.getConnection().getMetaData();
+    rsColumns = meta.getColumns(null, null, getTablename().toUpperCase(), null);
+    if (!rsColumns.isBeforeFirst()) {
+      // No Result Set, may be the table is stored in lower case
+      rsColumns = meta.getColumns(null, null, getTablename().toLowerCase(), null);
+    }
+    int totalColumns = columnNamesSet.size();
+    if (totalColumns == 0) {
+      //Direct mapping between POJO fields and DB columns
+      while (rsColumns.next()) {
```

Comment:
There is a code duplication here...
You could do following to avoid code duplication:

while (rsColumns.next()) {
  if (tableColumns == 0 || columnNamesSet.contains(columnname)) {
    // add columns to the set.
  }
}